src/test/java/com/aws/greengrass/detector/uploader/ConnectivityUpdaterTest.java

```diff
@@ -108,15 +108,23 @@ public void GIVEN_ips_changed_WHEN_get_ip_addresses_THEN_return_true() {
 
         // old ips null
         assertTrue(connectivityUpdater.hasIpsChanged(getIps()));
-        connectivityUpdater.setIpAddresses(getIps());
+        connectivityUpdater.setIpAddressesAndPort(getIps(), TestConstants.PORT_1);
         assertTrue(connectivityUpdater.hasIpsChanged(getNewIps()));
     }
 
     @Test
-    public void GIVEN_ips_not_changed_WHEN_get_ip_addresses_THEN_return_false() {
+    public void GIVEN_ips_and_port_not_changed_WHEN_has_ips_or_port_changed_THEN_return_false() {
         connectivityUpdater = new ConnectivityUpdater(deviceConfiguration,null);
-        connectivityUpdater.setIpAddresses(getIps());
-        assertFalse( connectivityUpdater.hasIpsChanged(getIps()));
+        connectivityUpdater.setIpAddressesAndPort(getIps(), TestConstants.PORT_1);
+        assertFalse(connectivityUpdater.hasIpsChanged(getIps()));
+        assertFalse(connectivityUpdater.hasPortChanged(TestConstants.PORT_1));
+    }
```

Comment:
This doesn't exercise any production code - all three of these methods are unit test only. It doesn't give us any confidence that the Updater is doing what it should be doing.

This should be tested by mocking the config and clientfactory objects and making sure that the public methods are doing what they're supposed to.